plugins/woocommerce/client/legacy/js/admin/marketplace-suggestions.js

```diff
@@ -109,11 +117,20 @@
 		}
 
 		// Render DOM element for suggestion linkout, optionally with button style.
-		function renderLinkout( context, product, promoted, slug, url, text, isButton ) {
+		function renderLinkout( context, product, promoted, slug, url, iamUrl, text, isButton ) {
 			var linkoutButton = document.createElement( 'a' );
-
 			var utmUrl = addURLParameters( context, url );
-			linkoutButton.setAttribute( 'href', utmUrl );
+			var isInternalLink = false;
+			var targetUrl = '';
+
+			if ( iamUrl ) {
+				targetUrl = getAdminLink( iamUrl );
+				isInternalLink = true;
+			} else {
+				targetUrl = utmUrl;
+			}
```

Comment:
I'd probably simplify it to:
```suggestion
			var isInternalLink = Boolean( iamUrl );
			var targetUrl = isInternalLink ? getAdminLink( iamUrl ) : utmUrl;
```

Although it's a personal preference, maybe if/else is more readable for some